### ISSUE TITLE
Dreddit-dapp issue

### ISSUE THREAD

**Issue by maxscience** [embark-framework/embark-docker#10]
I didn't have time to properly debug this but might still be useful to report.

1. Using Embark 3.1.10 via https://github.com/embark-framework/embark-docker
2. cd dreddit-dapp //(from https://github.com/status-im/dreddit-dapp ) 
3. run_embark
Error:
'Blockchain process ended before the end of this process. Code: 0'

It looks like Geth has some issues with this particular workflow. embark-docker runs just fine with other projects (the standard demo for example).

**Comment by jrainville** [comment]
Could you try to tun `embark blockchain`, to see what error message Geth tells you.

**Comment by michaelsbradleyjr** [comment]
@maxscience I'm able to run dreddit-dapp successfully using embark-docker. On what platform are you running docker? There is a bug that remains to be fixed when running the container with Docker for Windows, that's why I'm asking.

**Comment by maxscience** [comment]
@michaelsbradleyjr I'm running the latest version of Docker for Mac on Mac OS Mojave

@jrainville sure, here's the output:

Geth: WARN [10-09|16:46:11.953] Sanitizing cache to Go's GC limits       provided=1024 updated=666

Geth: INFO [10-09|16:46:11.955] Maximum peer count                       ETH=0 LES=0 total=0

Geth: INFO [10-09|16:46:12.846] Using developer account                  address=0x3b37cf1F4fBdDd735A762f04FB04CB62C9b8Fcf6

Geth: INFO [10-09|16:46:12.856] Starting peer-to-peer node               instance=Geth/v1.8.12-stable-37685930/linux-amd64/go1.10.3

Geth: INFO [10-09|16:46:12.857] Allocated cache and file handles         database=/dapp/.embark/development/datadir/geth/chaindata cache=499 handles=1024

Geth: INFO [10-09|16:46:12.910] Persisted trie from memory database      nodes=11 size=1.71kB time=74.416µs gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B

Geth: INFO [10-09|16:46:12.917] Initialised chain configuration          config="{ChainID: 1337 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: <nil> Engine: clique}"

Geth: INFO [10-09|16:46:12.917] Initialising Ethereum protocol           versions="[63 62]" network=1337

Geth: INFO [10-09|16:46:12.919] Loaded most recent local header          number=0 hash=0130bd…71950a td=1

Geth: INFO [10-09|16:46:12.920] Loaded most recent local full block      number=0 hash=0130bd…71950a td=1

Geth: INFO [10-09|16:46:12.921] Loaded most recent local fast block      number=0 hash=0130bd…71950a td=1

Geth: INFO [10-09|16:46:12.924] Loaded local transaction journal         transactions=0 dropped=0

Geth: INFO [10-09|16:46:12.931] Regenerated local transaction journal    transactions=0 accounts=0

Geth: INFO [10-09|16:46:12.932] Starting P2P networking 

Geth: INFO [10-09|16:46:12.932] RLPx listener up                         self="enode://4871533f620c6506d8dccf88131dcb4b1148a16b33305fde0ceef3bcb33247c08c7e713a39943bce2d02aaba063c29d51e99be98f6d74dd1e5ff28123e7b14ff@[::]:40541?discport=0"

Geth: INFO [10-09|16:46:12.932] started whisper v.6.0 

Geth: INFO [10-09|16:46:12.936] Blockchain manager stopped 

Geth: INFO [10-09|16:46:12.936] Stopping Ethereum protocol 
INFO [10-09|16:46:12.936] Ethereum protocol stopped 
INFO [10-09|16:46:12.937] Transaction pool stopped 

Geth: INFO [10-09|16:46:12.942] Database closed                          database=/dapp/.embark/development/datadir/geth/chaindata
INFO [10-09|16:46:12.942] whisper stopped 

Geth error: Fatal: Error starting protocol stack: listen unix /dapp/.embark/development/datadir/geth.ipc: bind: file name too long

Geth: Fatal: Error starting protocol stack: listen unix /dapp/.embark/development/datadir/geth.ipc: bind: file name too long

Geth exited with error code 1

**Comment by jrainville** [comment]
Ok, that's a known bug from Geth. When the file path is too long, it gives an error. 

However, in your case, it looks like your dapp path is only `/dapp` right?

If not, can you try to put your Dapp in a folder closer to the root.

**Comment by michaelsbradleyjr** [comment]
@jrainville the current working directory is automatically mounted to the container's `/dapp` by `run_embark`.

@maxscience did you start the blockchain with `run_embark blockchain`? I'm running Docker for Mac on macOS High Sierra. I have dreddit-dapp under `~/repos`. So to kick if off I'm doing:

```
source <(curl -L https://bit.ly/run_embark)
cd ~/repos/dreddit-dapp
run_embark blockchain
```

**Comment by maxscience** [comment]
@michaelsbradleyjr Yes that's what I did. Except my directory was under a longer path. I moved it under a shorter path as suggested by @jrainville and now it works 👍 
Looks like Geth was to blame then.

**Comment by michaelsbradleyjr** [comment]
It's interesting that the long path on the host can trigger the geth bug when that host-path is mounted as `/dapp`!

**Comment by jrainville** [comment]
Happy it works for you. Next time you see an error or the blockchain node exiting, remember to run `embark blockchain` to see more detailed information.

If you end up updating to Embark 3.2, you will see that we added a message for that, along with a multitude of new features.

**Comment by michaelsbradleyjr** [comment]
I was able to reproduce the problem by putting dreddit-dapp under a long/deep path on my macOS host.

Note that a container image for Embark 3.2 isn't available yet on Docker Hub, but it will be soon.
